app/models/repository/destroying.rb

```diff
@@ -1,30 +1,57 @@
 module Repository::Destroying
   extend ActiveSupport::Concern
 
+  DELETION_WAIT_TIME = 30.days
+
   included do
-    @destroying ||= []
-    before_destroy :mark_as_destroying, prepend: true
-  end
+    scope :destroying, ->() { unscoped.where(is_destroying: true) }
+    scope :active, ->() { where(is_destroying: false) }
 
-  def is_destroying?
-    self.class.instance_variable_get(:@destroying).include?(self.id)
+    def self.find_deleted_repository_with_owner(path, user)
+      repository = Repository.find_by_path(path)
+      repository if user.owned_ids('Repository').include?(repository.id)
+    end
   end
 
+  # Only use `destroy_asynchronously` if you want to destroy a repository.
+  # It prepares the deletion by setting a flag, which enables the deletion
+  # of its ontologies.
   def destroy
+    Rails.logger.info("Destroy #{self.class} #{self} (id: #{id})")
     super
   rescue StandardError => e
-    unmark_as_destroying
-    raise e.class, "Can't delete repository: It contains
-    #{Settings.OMS.with_indefinite_article} that is imported by another repository."
+    self.is_destroying = false
+    self.destroy_job_id = nil
+    self.destroy_job_at = nil
+    save!
+    raise e.class,
+          I18n.t('repository.delete_error',
+            oms: Settings.OMS.with_indefinite_article),
+          cause: e
   end
 
-  protected
+  def undestroy
+    Sidekiq::Status.unschedule(destroy_job_id)
+    self.is_destroying = false
+    self.destroy_job_id = nil
+    self.destroy_job_at = nil
+    save!
+  end
 
-  def mark_as_destroying
-    self.class.instance_variable_get(:@destroying) << self.id if self.id
+  def destroy_asynchronously
```

Comment:
Assignment Branch Condition size for destroy_asynchronously is too high. [15.03/15]